### PR TITLE
Disallow behavior editing if there is a toolbox with no categories

### PR DESCRIPTION
**Pull request by ajpal**
# Description
Fix for #32252, which was reverted by #32280 

Level with categorized toolbox (allows behavior editing):
s/coursef-2019/stage/3/puzzle/5
image

Level with no toolbox (allows behavior editing):
s/coursef-2019/stage/15/puzzle/2
image

Level with uncategorized toolbox (disallows behavior editing):
s/coursef-2019/stage/2/puzzle/4
image

## Links

- [jira](https://codedotorg.atlassian.net/browse/STAR-834)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by jmkulwik** [comment]
This seems like a workaround for a very specific use-case. Is there future work planned to allow levelbuilders to enable/disable the edit behavior button per level?

**Comment by ajpal** [comment]
> This seems like a workaround for a very specific use-case. Is there future work planned to allow levelbuilders to enable/disable the edit behavior button per level?

I don't think so at this point? This was the decision from the curriculum/eng meeting: https://docs.google.com/document/d/1EL99K31sSv3wwsfc5bNqtChh870oF3GepenWxyPm9Cg/edit#bookmark=id.swq3i21lhwkw

File: apps/src/p5lab/spritelab/blocks.js [hunk 1]
```diff
@@ -377,7 +378,12 @@ export default {
           .appendTitle(fieldLabel, 'VAR')
           .appendTitle(Blockly.Msg.VARIABLES_GET_TAIL);
 
-        if (Blockly.useModalFunctionEditor) {
+        let allowBehaviorEditing = Blockly.useModalFunctionEditor;
+        if (appOptions.level.toolbox && !Blockly.hasCategories) {
```
Comment (jmkulwik) [inline apps/src/p5lab/spritelab/blocks.js:385]: Can you add a comment explaining why we do this check? (No need to wait for drone after)

**Review by jmkulwik — APPROVED** [review]
Left one comment. Besides that, looks good!

**Pull request merged**